enhancements/optional-manifest.md

```diff
@@ -217,91 +218,32 @@ The logic in any future solution for bundle distribution, like RukPak, would nee
 
 ### Test Plan
 
-**Note:** *Section not required until targeted at a release.*
-
-Consider the following in developing a test plan for this enhancement:
-- Will there be e2e and integration tests, in addition to unit tests?
-- How will it be tested in isolation vs with other components?
-
-No need to outline all of the test cases, just the general strategy. Anything
-that would count as tricky in the implementation and anything particularly
-challenging to test should be called out.
+Code coverage for this enhancement is ensured by two set of unit tests.
 
-All code is expected to have adequate tests (eventually with coverage
-expectations).
+- Validation that a resource is properly identified as optional based on the Properties that have been configured.
+- Step processing:
+	- Validation that a step flagged as optional does not cause the failure of the InstallPlan if it cannot be successfully processed due to one of the API errors listed above as NotCreated.
+	- Validation that a step flagged as optional still fails the InstallPlan due to one of the API errors listed above as Failure.
+	- Validation that the logic has not changed for non optional steps
 
 ### Graduation Criteria
 
-**Note:** *Section not required until targeted at a release.*
-
-Define graduation milestones.
-
-These may be defined in terms of API maturity, or as something else. Initial proposal
-should keep this high-level with a focus on what signals will be looked at to
-determine graduation.
-
-Consider the following in developing the graduation criteria for this
-enhancement:
-- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
-- Deprecation
-
-Clearly define what graduation means.
-
-#### Examples
-
-These are generalized examples to consider, in addition to the aforementioned
-[maturity levels][maturity-levels].
-
-##### Dev Preview -> Tech Preview
+The optional manifest feature is currently scheduled to be available in the next release. It will first be provided as a tech-preview and begins to gather users' feedbacks.
```

Comment:
Tech preview features must follow a certain process (as defined [here](https://github.com/openshift/enhancements/blob/master/guidelines/techpreview.md#official-processmechanism-for-delivering-a-tp-feature)). Can you expand on this section a little bit to clarify the mechanisms we'll use for the feature gate?

Reply:
I am getting confused as you are pointing to downstream documentation. The main aspect there is the opt-in through a feature gate.
I did a quick search on the enhancement proposals and "generic constraint" is the only one I found for OLM that has this part covered. Looking at the code I see pkg/feature/feature.go but I have not found any use of it.
It is surely something I could look at but I am questioning the incentive for creating a precedent for a feature that does not introduce new CRDs and does not have any effect when not explicitly used. I understand that users can start depend on this feature without being aware of its maturity level and that is precisely the reason why I was proposing first to introduce it as a tech. preview. The API for this feature is "explicit bundle properties", which does not offer a standard way of versioning (following the CRD way I would have put it under "alpha" otherwise).
If that's the way to go I can just remove this part.

Reply:
Yep. Sorry disregard downstream references.

Side note: I think we should remove this `dev/tech preview` nomenclature from our upstream EP template, and make this section geared more toward introducing features behind generic upstream feature gates.

Reply:
@joelanford Thanks for the clarification. I have modified the verbiage and removed references to graduation, tech. preview and GA only keeping the fact that it won't be introduced behind a feature gate and that we will gather feedback on it.
Is it good to merge?

Reply:
To clarify, I think we should consider adding this feature behind a feature gate. That way we can get some feedback on it before we fully commit to it.

Reply:
pushed a new version with the feature gate. I was not expecting that a feature with such a low radius would be subject to it when as far as I have seen feature gates have not been used much with OLM.

Reply:
Yeah, IMO OLM's propensity for straight to GA has made things harder to maintain because we've delivered features under pressure in the past and have not had the time to collect feedback, which results in features we sometimes immediately regret. In the future, I think straight to GA should be the exception rather than the rule.

There are definitely some added maintenance burdens with feature gates, so its worth debating more. But my gut is that if we're able to do it:
- Our upstream selves will be able to deliver new features without having to worry too much about their impact downstream.
- Anyone maintaining OLM in downstreams can pick and choose features to enable and be less conservative when reviewing/proposing upstream features, knowing they can disable anything that they're unprepared to support.
- We'll build a better community because downstream vendor-specific RFEs will be implemented (at worst) behind upstream feature flags.

Reply:
@joelanford there are things we would need to discuss regarding feature gates, olm and personas but this is not the subject of this PR. As I have added the feature gate verbiage any other things you would like to see or is it now good to get merged from your point of view?

Reply:
Yep. Looks good to me. Thanks!